src/OpenTelemetry/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Diagnostics.Metrics;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Temporalio.Client;
+using Temporalio.Extensions.DiagnosticSource;
+using Temporalio.Extensions.OpenTelemetry;
+using Temporalio.Runtime;
+using Temporalio.Worker;
+using TemporalioSamples.OpenTelemetry;
+
+AssemblyName assemblyName = typeof(TemporalClient).Assembly.GetName();
+
+using var meter = new Meter(assemblyName.Name!, assemblyName.Version!.ToString());
+
+string instanceId = args.ElementAtOrDefault(0) ?? throw new ArgumentException("Must pass 'worker' or 'workflow' as the single argument");
+
+ResourceBuilder resourceBuilder = ResourceBuilder.
+    CreateDefault()
+    .AddService("TemporalioSamples.OpenTelemetry", serviceInstanceId: instanceId);
+
+using TracerProvider tracerProvider = Sdk.
+    CreateTracerProviderBuilder().
+    SetResourceBuilder(resourceBuilder).
+    AddSource(TracingInterceptor.ClientSource.Name, TracingInterceptor.WorkflowsSource.Name, TracingInterceptor.ActivitiesSource.Name).
+    AddOtlpExporter().
+    Build();
+
+using MeterProvider meterProvider = Sdk.
+    CreateMeterProviderBuilder().
+    SetResourceBuilder(resourceBuilder).
+    AddMeter(assemblyName.Name!).
+    AddOtlpExporter().
+    Build();
+
+// Create a client to localhost on default namespace
+var client = await TemporalClient.ConnectAsync(new("localhost:7233")
+{
+    LoggerFactory = LoggerFactory.Create(builder =>
+        builder.
+            AddSimpleConsole(options => options.TimestampFormat = "[HH:mm:ss] ").
+            SetMinimumLevel(LogLevel.Information)),
+    Interceptors = new[] { new TracingInterceptor() },
+    Runtime = new TemporalRuntime(new TemporalRuntimeOptions()
+    {
+        Telemetry = new TelemetryOptions()
+        {
+            Metrics = new MetricsOptions()
+            {
+                CustomMetricMeter = new CustomMetricMeter(meter),
```

Comment:
Hrmm, so there are two ways to configure OTel telemetry metrics. One way is this, but the other way is simply to set the OTel options in this metric options so our internal code forwards. We often encourage the latter because it's less setup and it matches other SDKs.

I wonder if there is some way we can show both? Like maybe switched from one to the other via CLI flag?

Reply:
I split this into two samples,  `OpenTelemetry/CoreSdkFowarding` and `OpenTelemetry/DotNetMetrics`.

I thought this was cleanest, and follows a similar pattern that exists with the polling samples

Reply:
Ok, will look. Alternatively we probably could have just had a different parameter accepted in the `Program.cs` to choose one approach or the other.